pwiz_tools/Skyline/EditUI/PasteDlg.cs

```diff
@@ -444,57 +459,48 @@ private SrmDocument AddPeptides(SrmDocument document, bool validating, ref Ident
             return document;
         }
 
-        private List<string> ListPeptideSequences()
+        /// <summary>
+        /// Returns a list of tuples with the peptide sequences and optional charges.
+        /// The returned list may contain nulls, and will have exactly the same number of rows as gridViewPeptides.
+        /// </summary>
+        private List<Tuple<string, Adduct>> ListPeptideSequences()

```

Comment:
I really hate Tuple, especially as a return value like this. It makes the calling code with Item1 and Item2 require a look at the function returning the Tuple to know what these are. I much prefer defining a struct with named properties.